lib/scala/editions/src/main/scala/org/enso/editions/EditionName.scala

```diff
@@ -29,9 +29,15 @@ object EditionName {
   implicit val editionNameDecoder: Decoder[EditionName] = { json =>
     json
       .as[String]
-      .orElse(json.as[Int].map(_.toString))
-      .orElse(json.as[Float].map(_.toString))
-      .map(EditionName(_))
+      .fold(
+        _ =>
+          if (json.value == Json.Null)
+            Left(DecodingFailure("edition cannot be empty", Nil))
+          else
+            json.as[Int].orElse(json.as[Float]),
```

Comment:
Version can only have a single `.`? Semantic versioning usually expects three digits and two dots: `1.2.3`

Reply:
That is true. Seems like a yet another bug. @radeusgd mentioned that this was introduced only for convenience, if someone writes `1.2` rather than `"1.2"`. But yes, `1.2.3` wouldn't work.

Reply:
The version is defined to be more or less an arbitrary string.

The thing is, the YAML format, or at least the circe parse we use is **very** weird.

IIRC if it sees:
```
version: 1.2.3
```
it will not match a 'numeric' pattern, so it will be treated as `String` - `"1.2.3"` and we will parse it correctly by the very first `as[String]`.

The issue is when the version is like:
```
version: 2023.2
```
which is what our edition names were planned to be.

Now circe sees `x.y` and thinks - hmm these look like numbers. And it parses it as a float. And so our first `as[String]` fails... So then this `as[Float]` exists solely as a fallback to be able to still parse such values.

It is a horrible hack and I cannot wait until we parse `12345.2` as `1234.199999999999` or something... I guess ideally we should somehow force circe to treat this as a string, but I'm not sure if that is possible - back in the day I just settled for this workaround as it worked well enough for the cases we had - but it's definitely not a perfect solution.